### PR TITLE
Use a tag to declare the order of children

### PR DESCRIPTION
**Pull request by panglesd**
Currently, it is:

```
@children_order foo    bar dir/ baz
             boo
```

(The tag expect a single paragraph containing only words and whitespace)

Based on #1239.

### CODE REVIEW

File: src/parser/test/test.ml [hunk 1]
```diff
@@ -143,6 +143,8 @@ module Ast_to_sexp = struct
           @ List.map (at.at (nestable_block_element at)) es)
     | `Return es ->
         List (Atom "@return" :: List.map (at.at (nestable_block_element at)) es)
+    | `Children_order es ->
+        List (Atom "@return" :: List.map (at.at (nestable_block_element at)) es)
```
Comment (gpetiot) [inline src/parser/test/test.ml:147]:
```suggestion
        List (Atom "@children_order" :: List.map (at.at (nestable_block_element at)) es)
```

File: src/model/semantics.ml [hunk 2]
```diff
@@ -57,6 +70,24 @@ let handle_internal_tags (type a) tags : a handle_internal_tags -> a = function
           None
       | Some ((`Dot _ as p), _) -> Some p
       | None -> None)
+  | Expect_page_tags ->
+      let unparsed_lines =
+        find_tags []
+          (function `Children_order _ as p -> Some p | _ -> None)
```
Comment (gpetiot) [inline src/model/semantics.ml:76]: I think `find_tags` would be easier to read if `f` was labelled:
```suggestion
          ~filter:(function `Children_order _ as p -> Some p | _ -> None)
```

Comment (jonludlam) [reply src/model/semantics.ml]: Now `find_tags` is inconsistent with `find_tag` immediately above it, which still takes an `f` parameter :-P

Comment (panglesd) [reply src/model/semantics.ml]: Thanks!

**Review by gpetiot — APPROVED** [review]
Looks good now!

**Review by Julow — APPROVED** [review]
Looks good :) I particularly like that the frontmatter tags don't end up in the model.

**Comment by jonludlam** [comment]
Thanks!

**Pull request merged**